src/pudl/transform/eia930.py

```diff
@@ -0,0 +1,179 @@
+    ]
+
+    # Rename columns so that they contain only the energy source and the level of
+    # processing, so it's easy to construct a multi-index to unstack below
+    # Doing these manipulations while the values are in the column names rather than
+    # after they've been turned into a categorical column with millions of entries
+    # is much faster.
+    netgen_renamed = netgen.rename(
+        lambda col: col.removeprefix("net_generation_").removesuffix("_mw"),
+        axis="columns",
+    ).set_index(qual_cols)
+
+    # Prepare a multi-index for the columns so that we can stack cleanly
+    netgen_renamed.columns = pd.MultiIndex.from_tuples(
+        [x.split("_") for x in netgen_renamed.columns], names=["energy_source", None]
+    )
+    netgen_stacked = (
+        netgen_renamed.stack(level=0, future_stack=True)
+        .rename(columns=lambda x: f"net_generation_{x}_mw")
+        .reset_index()
+        .astype({"energy_source": "string"})
+    )
+    # TODO[zaneselvans] 2024-04-20: Verify that sum of net generation from all fuels
+    # adds up to the total And then drop the total rows.
+    # NOTE: currently there are some BIG differences between the calculated total and
+    # the reported total.
+
+    # netgen_stacked = netgen_stacked[netgen_stacked["energy_source"] != "total"
+
+    demand = raw_eia930__balance[
+        qual_cols + list(raw_eia930__balance.filter(like="demand"))
+    ]
+
+    return (
+        Output(
+            value=netgen_stacked,
+            output_name="core_eia930__hourly_balancing_authority_net_generation",
+        ),
+        Output(
+            value=demand,
+            output_name="core_eia930__hourly_balancing_authority_demand",
+        ),
+    )
+
+
+@multi_asset(
+    outs={
+        "core_eia930__hourly_subregion_demand": AssetOut(
+            io_manager_key="parquet_io_manager"
+        ),
+        "core_eia930__assn_balancing_authority_subregion": AssetOut(
+            io_manager_key="pudl_io_manager"
+        ),
+    },
+    compute_kind="pandas",
+)
+def core_eia930__hourly_subregion_assets(raw_eia930__subregion: pd.DataFrame):
+    """Produce a normalized table of hourly demand by subregion."""
+    demand = raw_eia930__subregion.assign(
+        subregion_code_eia=lambda df: df["subregion_code_eia"].str.upper()
+    ).loc[
+        :,
+        [
+            "report_datetime_local",
+            "report_datetime_utc",
+            "balancing_authority_code_eia",
+            "subregion_code_eia",
+            "demand_reported_mw",
+        ],
+    ]
+    assn = (
+        demand.groupby("balancing_authority_code_eia")["subregion_code_eia"]
+        .unique()
+        .explode()
+        .to_frame()
+        .reset_index()
+    )
+    return (
+        Output(value=demand, output_name="core_eia930__hourly_subregion_demand"),
+        Output(
+            value=assn, output_name="core_eia930__assn_balancing_authority_subregion"
+        ),
+    )
```

Comment:
instead of returning two assets here you could just make `core_eia930__hourly_subregion_demand` and then make `core_eia930__assn_balancing_authority_subregion` take `core_eia930__hourly_subregion_demand` as an input. I think that would be slightly more readable.

Reply:
We could, but they're really both derived from the one input, and creating them at the same time means that we don't have to read the other table back in again after it's been produced.